File: config/config-sbc2ha-new/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/sensor/oneWire/therm/OneWireTherm.java
package com.dfi.sbc2ha.config.sbc2ha.definition.sensor.oneWire.therm;

import com.dfi.sbc2ha.config.sbc2ha.definition.sensor.ThermBusSensorConfig;
import com.dfi.sbc2ha.helper.deserializer.BigIntegerSerializer;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigInteger;

@EqualsAndHashCode(callSuper = true)
@Data
public abstract class OneWireTherm extends ThermBusSensorConfig {

    @JsonSerialize(using = BigIntegerSerializer.class)
    BigInteger address;

    public OneWireTherm() {
        super();
    }
}

File: config/editor/src/app/definition/sensor/oneWire/therm/OneWireTherm.ts
import {ThermBusSensorConfig} from "../../ThermBusSensorConfig";
import {PlatformType} from "../../../enums/PlatformType";


export abstract class OneWireTherm extends ThermBusSensorConfig {
    address: string;

    constructor(platform: PlatformType, name: string, busId: string, address: string) {
        super(platform, name, busId);
        this.address = address;
    }
}
